### DISCUSSION TITLE
Source Salesforce: allow to have conditional queries

### DISCUSSION THREAD

**Discussion by DanielTrontsios** [airbytehq/airbyte#31266] [Connector Ideas and Features]
Hello, Airbyte community,

After setting up a few connectors with Salesforce I wasn't able to find a way to make conditional queries to my org. Is there a way to achieve that?

**Comment by marcosmarxm** [comment]
Today this is not possible. Airbyte has the option to filter streams to reduce the payload returned during the discover step and make the connector faster.
<img width="657" alt="image" src="https://github.com/airbytehq/airbyte/assets/5154322/44e959f0-8f03-48ce-9a57-356de28b28ad">


Airbyte connectors always try to retrieve all records and you need to filter downstream. This is helpful if you need this data in the future and follows the ELT paradigm. Sometimes, because you have a huge data probably you want to have ways to filter or reduce the ingestion... I changed your discussion to be an Idea and let's see what other users think about.

**Comment by PilarIH** [comment]
I'm looking for a way to specify the tables that you'd need to ingest from Salesforce without having to create a connector per each table or having an issue if I create a new table in SF that would comply with the filter but I don't want it in the E/L pipeline. Is there any solution for this so far?

[a comment by an automated account is not shown]
